lizmap/modules/lizmap/lib/Logger/Config.php

```diff
@@ -0,0 +1,156 @@
+        'active',
+        'profile',
+    );
+
+    // list of the logItems of the ini file
+    protected $logItems = array();
+    // If the log is active globally or not
+    private $active = '';
+
+    // database profile
+    private $profile = '';
+
+    protected $appContext;
+
+    public function __construct($readConfigPath, App\AppContextInterface $appContext)
+    {
+        $this->data = $readConfigPath;
+        $this->appContext = $appContext;
+
+        // set generic parameters
+        foreach ($this->properties as $prop) {
+            if (isset($readConfigPath['general'][$prop])) {
+                $this->{$prop} = $readConfigPath['general'][$prop];
+            }
+        }
+    }
+
+    public function getProperties()
+    {
+        return $this->properties;
+    }
+
+    /**
+     * Get a log item.
+     *
+     * @param string $key Key of the log item to get
+     *
+     * @return lizmapLogItem
+     */
+    public function getLogItem($key)
+    {
+        if (!key_exists($key, $this->logItems)) {
+            if (!key_exists('item:'.$key, $this->data)) {
+                return null;
+            }
+            $this->logItems[$key] = new Item($key, $this->data['item:'.$key], $this->appContext);
+        }
+
+        return $this->logItems[$key];
+    }
+
+    /**
+     * Get a list of log items names.
+     *
+     * @return string[] list of names
+     */
+    public function getLogItemList()
+    {
+        $logItemList = array();
+
+        foreach ($this->data as $section => $data) {
+            $match = preg_match('#(^item:)#', $section, $matches);
+            if (isset($matches[0])) {
+                $logItemList[] = str_replace($matches[0], '', $section);
+            }
+        }
+
+        return $logItemList;
+    }
+
+    /**
+     * Modify the general options.
+     *
+     * @param array $data array containing the global config data
+     */
+    public function modify($data)
+    {
+        $modified = false;
+        if (!$data) {
+            return $modified;
+        }
+        foreach ($data as $k => $v) {
+            if (in_array($k, $this->properties)) {
+                $this->data['general'][$k] = $v;
+                $this->{$k} = $v;
+                $modified = true;
+            }
+        }
+
+        return $modified;
+    }
+
+    /**
+     * Update the global config data. (modify and save).
+     *
+     * @param array      $data array containing the data of the general options
+     * @param null|mixed $ini
+     */
+    public function update($data, $ini = null)
+    {
+        $modified = $this->modify($data);
+        if ($modified) {
+            $modified = $this->save($ini);
+        }
+
+        return $modified;
+    }
+
+    /**
+     * save the global configuration data.
+     *
+     * @param null|mixed $ini
+     */
+    public function save($ini = null)
+    {
+        if (!$ini) {
+            $iniFile = $this->appContext->appConfigPath('lizmapLogConfig.ini.php');
```

Comment:
I think it is better if we give the ini filename to the constructor, we could test it easily